cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceQueryAdmin.java

```diff
@@ -103,11 +102,6 @@ public NamespaceMeta get(Id.Namespace namespaceId) throws Exception {
    */
   @Override
   public boolean exists(Id.Namespace namespaceId) throws Exception {
-    try {
-      get(namespaceId);
-    } catch (NotFoundException e) {
-      return false;
-    }
-    return true;
+    return nsStore.get(namespaceId) != null;
```

Comment:
The client and the server implements the same interface, hence the `exists()` is there. About the return code of `get()`, I notice one thing interesting. If the entity is not there, then it throws not found, however, if the entity is there but the user is not authorized to get it, then it throws unauthorized. Not sure if it is a security breach or not (since an unauthorized user can gain information about whether the entity is there or not).